core/lib/src/fs/rewrite.rs

```diff
@@ -72,8 +72,14 @@ impl<'r> File<'r> {
         }
     }
 
-    /// Returns `true` if the file is a dotfile. A dotfile is a file whose name
-    /// starts with a period (`.`) and is considered hidden.
+    /// Returns `true` if the file is a dotfile. A dotfile is a file whose
+    /// name or any directory in it's path start with a period (`.`) and is
+    /// considered hidden.
+    ///
+    /// # Windows Note
+    ///
+    /// This does *not* check the file metadata on any platform, so hidden files
+    /// on Windows will not be detected.
```

Comment:
This is bit more complex than the dotfiles check. First, it would only work after `Prefix` (much like `is_dir` or `exists`), and the current dotfiles check is typically done before `Prefix`.

Second, to match the behavior of dotfiles, we would likely want to also check all parent folders, which would likely represent a non-trivial amount of work. Right now, each `Rewrite` requires at most one syscall, and I think that's a reasonable limit.

Overall, I'm not sure if it's worthwhile. We could provide an alternate method that does check fs metadata, but I'm not sure how much value it would provide. We would also likely need to create a more complex test for this functionality, since I'm not sure git preserves Windows file properties.